### PR TITLE
Cw 514 add sort functionality for addressbook  mywallets and contacts

### PR DESCRIPTION
**Pull request by Serhii-Borodenko**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: lib/store/settings_store.dart [hunk 1]
```diff
@@ -324,14 +328,24 @@ abstract class SettingsStoreBase with Store {
...
+
     reaction(
         (_) => walletListAscending,
         (bool walletListAscending) =>
             sharedPreferences.setBool(PreferencesKey.walletListAscending, walletListAscending));
 
+    reaction(
+            (_) => contactListAscending,
+            (bool contactListAscending) =>
+            sharedPreferences.setBool(PreferencesKey.walletListAscending, contactListAscending));
```
[the remark this answers is not in the document]
Comment (Serhii-Borodenko) [reply lib/store/settings_store.dart:347]: fixed

File: lib/entities/contact.dart [hunk 2]
```diff
@@ -25,6 +26,9 @@ class Contact extends HiveObject with Keyable {
   @HiveField(2, defaultValue: 0)
   late int raw;
 
+  @HiveField(3)
+  late DateTime lastChange;
```
[the remark this answers is not in the document]
Comment (Serhii-Borodenko) [reply lib/entities/contact.dart]: double-checked lastChange initialization in the constructor. It’s works for me, I removed late and moved it to the initializer for clarity.

**Review by OmarHatem28 — APPROVED** [review]
Looks good, will just wait tests from Ramo

File: lib/src/screens/contact/contact_list_page.dart [hunk 3]
```diff
@@ -77,56 +77,101 @@ class ContactListPage extends BasePage {
...
+    return ListView.builder(
+      shrinkWrap: true,
+      itemCount: walletContacts.length * 2,
+      itemBuilder: (context, index) {
+        if (index.isOdd) {
+          return StandardListSeparator();
+        } else {
+          final walletInfo = walletContacts[index ~/ 2];
+          return generateRaw(context, walletInfo);
+        }
```
Comment (OmarHatem28) [inline lib/src/screens/contact/contact_list_page.dart:172]: for future instances, there is `ListView.separated` where it will do this automatically

**Review by OmarHatem28 — APPROVED** [review]

File: lib/src/screens/contact/contact_list_page.dart (hunk 3 above)
Comment (Serhii-Borodenko) [reply lib/src/screens/contact/contact_list_page.dart:172]: In this specific case, we need a separator at the bottom of the list. ListView.separated only provides a separator between items.

**Pull request merged**